### PR TITLE
Cleaning MessageListFragment

### PR DESCRIPTION
**Pull request by philipwhiuk**
MessageListFragment is one of the messiest classes in the code base.

It does way way too much.

* Interacts with EmailProvider, querying columns. To do this it has projection stuff.
* Formats the UI for the message list (so it is writing spans and stuff)
* Much more.

I've not tried to unpick the whole lot in one review. That would a nightmare.

This review is split into two commits:

The first

* Extracts the projection statics into a new class, MLFProjectionInfo
* Extracts some methods with no side-affects into MLFUtils
* Moves code out of the massive MessageListAdapter.bindView method into helper functions

The second:

Pulls the MessageListAdapter and MessageViewHolder out entirely.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java [hunk 1]
```diff
@@ -1144,42 +971,35 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+                        MLFUtils.setLastSelectedFolderName(mPreferences, messages, destFolderName);
                     }
-                }
 
-                switch (requestCode) {
-                case ACTIVITY_CHOOSE_FOLDER_MOVE:
-                    move(messages, destFolderName);
-                    break;
+                    switch (requestCode) {
+                    case ACTIVITY_CHOOSE_FOLDER_MOVE:
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java]: indentation

File: k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java [hunk 2]
```diff
@@ -1924,10 +1640,121 @@ public void bindView(View view, Context context, Cursor cursor) {
...
+         * Create a span section for the sender, and assign the correct font size and weight
+         */
+        private AbsoluteSizeSpan buildSenderSpan() {
+            int fontSize = (mSenderAboveSubject) ?
+                    mFontSizes.getMessageListSubject():
+                    mFontSizes.getMessageListSender();
+            return new AbsoluteSizeSpan(fontSize, true);
+        }
+
+        private Address fetchCounterPartyAddress(boolean fromMe, Address[] toAddrs, Address[] ccAddrs, Address[] fromAddrs) {
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java]: this is more of a "get" than a "fetch", isn't it?

File: k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java [hunk 3]
```diff
@@ -1924,10 +1640,121 @@ public void bindView(View view, Context context, Cursor cursor) {
                 holder.subject.setTypeface(Typeface.create(holder.subject.getTypeface(), maybeBoldTypeface));
                 holder.subject.setText(subject);
             }
-
             holder.date.setText(displayDate);
         }
 
+        private void formatPreviewText(TextView preview, CharSequence beforePreviewText, String sigil) {
+            Spannable previewText = (Spannable)preview.getText();
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java]: missing space

File: k9mail/src/main/java/com/fsck/k9/fragment/MLFUtils.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import com.fsck.k9.mail.Folder;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mailstore.LocalFolder;
+import com.fsck.k9.mailstore.LocalStore;
+import timber.log.Timber;
+
+import static com.fsck.k9.fragment.MLFProjectionInfo.SENDER_LIST_COLUMN;
+
+
+public class MLFUtils {
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/fragment/MLFUtils.java]: I think this should be `MlfUtils` per code style guidelines.

File: k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java [hunk 5]
```diff
@@ -215,9 +179,9 @@ public static MessageListFragment newInstance(LocalSearch search, boolean isThre
         SORT_COMPARATORS = Collections.unmodifiableMap(map);
     }
 
-    private ListView mListView;
+    ListView mListView;
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java]: might as well get rid of the m prefixes while you're doing cleanup

**Comment by philipwhiuk** [comment]
Rebased. Merging.

**Pull request merged**